examples/elmoControl/elmoControl.cc

```diff
@@ -0,0 +1,183 @@
+        printf("usage no redundancy mode : ./test NIC_nominal\n");
+        return 1;
+    }
+
+
+    std::shared_ptr<AbstractSocket> socketRedundancy;
+    std::string red_interface_name = "null";
+    std::string nom_interface_name = argv[1];
+
+    if (argc == 2)
+    {
+        printf("No redundancy mode selected \n");
+        socketRedundancy = std::make_shared<SocketNull>();
+    }
+    else
+    {
+        socketRedundancy = std::make_shared<Socket>();
+        red_interface_name = argv[2];
+    }
+
+    auto socketNominal = std::make_shared<Socket>();
+    try
+    {
+        socketNominal->open(nom_interface_name);
+        socketRedundancy->open(red_interface_name);
+    }
+    catch (std::exception const& e)
+    {
+        std::cerr << e.what() << std::endl;
+        return 1;
+    }
+
+    auto reportRedundancy = []()
+    {
+        printf("Redundancy has been activated due to loss of a cable \n");
+    };
+
+    std::shared_ptr<Link> link= std::make_shared<Link>(socketNominal, socketRedundancy, reportRedundancy);
+    link->setTimeout(2ms);
+    link->checkRedundancyNeeded();
+
+    Bus bus(link);
+
+    uint8_t io_buffer[2048];
+    try
+    {
+        bus.init();
+
+        // Prepare mapping for elmo
+        uint32_t rxLength = sizeof(pdo::RxMapping);
+        bus.writeSDO(bus.slaves().at(0), sdo::RxPDO.index, sdo::RxPDO.subindex, 
+                    Bus::Access::COMPLETE, const_cast<uint8_t*>(pdo::RxMapping), rxLength);
+        uint32_t txLength = sizeof(pdo::TxMapping);
+        bus.writeSDO(bus.slaves().at(0), sdo::TxPDO.index, sdo::TxPDO.subindex, 
+                    Bus::Access::COMPLETE, const_cast<uint8_t*>(pdo::TxMapping), txLength);
+        
+        bus.createMapping(io_buffer);
+
+        bus.requestState(State::SAFE_OP);
+        bus.waitForState(State::SAFE_OP, 1s);
+    }
+    catch (ErrorCode const& e)
+    {
+        std::cerr << e.what() << ": " << ALStatus_to_string(e.code()) << std::endl;
+        return 1;
+    }
+    catch (std::exception const& e)
+    {
+        std::cerr << e.what() << std::endl;
+        return 1;
+    }
+
+    auto callback_error = [](DatagramState const&){ THROW_ERROR("something bad happened"); };
+    auto false_alarm = [](DatagramState const&){ DEBUG_PRINT("previous error was a false alarm"); };
+
+    try
+    {
+        bus.processDataRead(callback_error);
+        bus.processDataWrite(false_alarm);
+    }
+    catch (...)
+    {
+    }
+
+    try
+    {
+        bus.requestState(State::OPERATIONAL);
+        bus.waitForState(State::OPERATIONAL, 100ms);
+    }
+    catch (ErrorCode const& e)
+    {
+        std::cerr << e.what() << ": " << ALStatus_to_string(e.code()) << std::endl;
+        return 1;
+    }
+    catch (std::exception const& e)
+    {
+        std::cerr << e.what() << std::endl;
+        return 1;
+    }
+
+    link->setTimeout(500us);
+
+    Slave& elmo = bus.slaves().at(0);
+    pdo::Output* outputPDO = reinterpret_cast<pdo::Output*>(elmo.output.data);
```

Comment:
snake case (maybe remove PDO)

Reply:
namespace pdo prevents mixup with enum kickcat::SyncManagerType (containing Output and Input)